x-pack/plugins/spaces/server/routes/api/external/delete.ts

```diff
@@ -33,6 +34,13 @@ export function initDeleteSpacesApi(deps: ExternalRouteDeps) {
       } catch (error) {
         if (SavedObjectsClient.errors.isNotFoundError(error)) {
           return response.notFound();
+        } else if (SavedObjectsClient.errors.isEsCannotExecuteScriptError(error)) {
+          log.error(
+            `Failed to delete space '${id}', cannot execute script in Elasticsearch query: ${error.message}`
+          );
+          return response.customError(
+            wrapError(Boom.notImplemented('Cannot execute script in Elasticsearch query'))
+          );
```

Comment:
I tested this manually (by starting Elasticsearch with `script.allowed_types='none'`, and again with `script.allowed_contexts='aggs'`). Also added a unit test to exercise this functionality. I thought an integration test was overkill, though, as we'd need a different test suite to do it.